### PR TITLE
Getting Rid of Swing

### PR DESCRIPTION
**Pull request by LinusDietz**
Since we made good progress in removing Swing so far, I created an architecture test to check if Swing and JGoodies is still used.

Also, I could delete quite some unused classes.

Can only be merged, if Swing is entirely removed.

### CODE REVIEW

**Comment by Siedlerchr** [comment]
General things: Undo/Redo

Classes which need some further conversion;
EmacsKeybinding (does anyone use that?)
  MergeSharedEntryDialog  (we do have a merge dialog, can we reuse it?)
org.jabref.gui.importer.actions.CheckForNewEntryTypesAction.createCheckBoxPane

GroupTreeNodeViewModel => Can this be removed or converted?

FileListTableModel -> can be removed, basePanel method must be adjusted to linkedFiles dialog

**Comment by LinusDietz** [comment]
To move on with this, I have excluded the UndoManager from the checks. However, we still have 90 violations.

<details><summary>Test log</summary>
java.lang.AssertionError: Architecture Violation [Priority: MEDIUM] - Rule 'no classes should access classes that reside in any package ['javax.swing', 'javax.swing.border..', 'javax.swing.colorchooser..', 'javax.swing.event..', 'javax.swing.filechooser..', 'javax.swing.plaf..', 'javax.swing.table..', 'javax.swing.text..', 'javax.swing.tree..']' was violated (90 times):
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls constructor <javax.swing.DefaultListModel.<init>()> in (FieldSetComponent.java:83)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls constructor <javax.swing.JButton.<init>(java.lang.String)> in (FieldSetComponent.java:81)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls constructor <javax.swing.JButton.<init>(java.lang.String)> in (FieldSetComponent.java:82)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls constructor <javax.swing.JButton.<init>(javax.swing.Icon)> in (FieldSetComponent.java:125)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls constructor <javax.swing.JButton.<init>(javax.swing.Icon)> in (FieldSetComponent.java:126)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls constructor <javax.swing.JComboBox.<init>([Ljava.lang.Object;)> in (FieldSetComponent.java:162)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls constructor <javax.swing.JLabel.<init>(java.lang.String)> in (FieldSetComponent.java:86)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls constructor <javax.swing.JList.<init>(javax.swing.ListModel)> in (FieldSetComponent.java:92)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls constructor <javax.swing.JPanel.<init>()> in (FieldSetComponent.java:79)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls constructor <javax.swing.JScrollPane.<init>(java.awt.Component, int, int)> in (FieldSetComponent.java:117)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls constructor <javax.swing.JTextField.<init>(int)> in (FieldSetComponent.java:157)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls method <javax.swing.Box.createHorizontalStrut(int)> in (FieldSetComponent.java:138)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls method <javax.swing.DefaultListModel.addElement(java.lang.Object)> in (FieldSetComponent.java:90)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls method <javax.swing.JButton.addActionListener(java.awt.event.ActionListener)> in (FieldSetComponent.java:104)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls method <javax.swing.JButton.addActionListener(java.awt.event.ActionListener)> in (FieldSetComponent.java:127)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls method <javax.swing.JButton.addActionListener(java.awt.event.ActionListener)> in (FieldSetComponent.java:128)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls method <javax.swing.JButton.addActionListener(java.awt.event.ActionListener)> in (FieldSetComponent.java:95)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls method <javax.swing.JButton.setToolTipText(java.lang.String)> in (FieldSetComponent.java:129)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls method <javax.swing.JButton.setToolTipText(java.lang.String)> in (FieldSetComponent.java:130)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls method <javax.swing.JComboBox.setEditable(boolean)> in (FieldSetComponent.java:163)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls method <javax.swing.JList.addFocusListener(java.awt.event.FocusListener)> in (FieldSetComponent.java:175)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls method <javax.swing.JList.setSelectionMode(int)> in (FieldSetComponent.java:93)
Constructor <org.jabref.gui.customentrytypes.FieldSetComponent.<init>(java.lang.String, java.util.List, java.util.List, java.lang.String, java.lang.String, boolean, boolean)> calls method <javax.swing.JTextField.addActionListener(java.awt.event.ActionListener)> in (FieldSetComponent.java:158)
Constructor <org.jabref.gui.search.SearchWorker.<init>(org.jabref.gui.BasePanel, org.jabref.logic.search.SearchQuery, org.jabref.gui.search.SearchDisplayMode)> calls constructor <javax.swing.SwingWorker.<init>()> in (SearchWorker.java:32)
Constructor <org.jabref.gui.shared.MergeSharedEntryDialog.<init>(org.jabref.gui.JabRefFrame, org.jabref.model.database.shared.DatabaseSynchronizer, org.jabref.model.entry.BibEntry, org.jabref.model.entry.BibEntry, org.jabref.model.database.BibDatabaseMode)> calls constructor <javax.swing.JDialog.<init>(java.awt.Frame, java.lang.String, boolean)> in (MergeSharedEntryDialog.java:43)
Method <org.jabref.gui.BasePanel.markBaseChanged()> calls method <javax.swing.SwingUtilities.invokeAndWait(java.lang.Runnable)> in (BasePanel.java:977)
Method <org.jabref.gui.BasePanel.markBaseChanged()> calls method <javax.swing.SwingUtilities.isEventDispatchThread()> in (BasePanel.java:973)
Method <org.jabref.gui.JabRefFrame$1.run()> calls method <javax.swing.SwingUtilities.invokeLater(java.lang.Runnable)> in (JabRefFrame.java:295)
Method <org.jabref.gui.customentrytypes.FieldSetComponent$FieldListFocusListener.focusGained(java.awt.event.FocusEvent)> calls method <javax.swing.JList.getSelectedValue()> in (FieldSetComponent.java:340)
Method <org.jabref.gui.customentrytypes.FieldSetComponent$FieldListFocusListener.focusGained(java.awt.event.FocusEvent)> calls method <javax.swing.JList.setSelectedIndex(int)> in (FieldSetComponent.java:341)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.addField(java.lang.String)> calls method <javax.swing.DefaultListModel.contains(java.lang.Object)> in (FieldSetComponent.java:238)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.addField(java.lang.String)> calls method <javax.swing.JOptionPane.showMessageDialog(java.awt.Component, java.lang.Object, java.lang.String, int)> in (FieldSetComponent.java:246)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.addFieldUncritically(java.lang.String)> calls method <javax.swing.DefaultListModel.addElement(java.lang.Object)> in (FieldSetComponent.java:260)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.addListDataListener(javax.swing.event.ListDataListener)> calls method <javax.swing.DefaultListModel.addListDataListener(javax.swing.event.ListDataListener)> in (FieldSetComponent.java:307)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.addListSelectionListener(javax.swing.event.ListSelectionListener)> calls method <javax.swing.JList.addListSelectionListener(javax.swing.event.ListSelectionListener)> in (FieldSetComponent.java:295)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.getFields()> calls method <javax.swing.DefaultListModel.elements()> in (FieldSetComponent.java:222)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.getFields()> calls method <javax.swing.DefaultListModel.getSize()> in (FieldSetComponent.java:221)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.getFirstSelected()> calls method <javax.swing.JList.getSelectedValue()> in (FieldSetComponent.java:198)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.lambda$new$0(java.awt.event.ActionEvent)> calls method <javax.swing.JComboBox.getSelectedItem()> in (FieldSetComponent.java:97)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.lambda$new$0(java.awt.event.ActionEvent)> calls method <javax.swing.JComboBox.getSelectedItem()> in (FieldSetComponent.java:98)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.lambda$new$0(java.awt.event.ActionEvent)> calls method <javax.swing.JTextField.getText()> in (FieldSetComponent.java:100)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.lambda$new$0(java.awt.event.ActionEvent)> calls method <javax.swing.JTextField.getText()> in (FieldSetComponent.java:101)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.lambda$new$4(java.awt.event.ActionEvent)> calls method <javax.swing.JTextField.getText()> in (FieldSetComponent.java:158)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.move(int)> calls method <javax.swing.DefaultListModel.add(int, java.lang.Object)> in (FieldSetComponent.java:323)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.move(int)> calls method <javax.swing.DefaultListModel.get(int)> in (FieldSetComponent.java:319)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.move(int)> calls method <javax.swing.DefaultListModel.remove(int)> in (FieldSetComponent.java:322)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.move(int)> calls method <javax.swing.DefaultListModel.size()> in (FieldSetComponent.java:321)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.move(int)> calls method <javax.swing.JList.getSelectedIndex()> in (FieldSetComponent.java:315)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.move(int)> calls method <javax.swing.JList.setSelectedIndex(int)> in (FieldSetComponent.java:324)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.removeSelected()> calls method <javax.swing.DefaultListModel.removeElementAt(int)> in (FieldSetComponent.java:274)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.removeSelected()> calls method <javax.swing.JList.getSelectedIndices()> in (FieldSetComponent.java:269)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.selectField(java.lang.String)> calls method <javax.swing.DefaultListModel.indexOf(java.lang.Object)> in (FieldSetComponent.java:183)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.selectField(java.lang.String)> calls method <javax.swing.JList.getCellBounds(int, int)> in (FieldSetComponent.java:190)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.selectField(java.lang.String)> calls method <javax.swing.JList.setSelectedIndex(int)> in (FieldSetComponent.java:185)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.selectField(java.lang.String)> calls method <javax.swing.JScrollPane.getViewport()> in (FieldSetComponent.java:189)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.selectField(java.lang.String)> calls method <javax.swing.JViewport.scrollRectToVisible(java.awt.Rectangle)> in (FieldSetComponent.java:192)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.setEnabled(boolean)> calls method <javax.swing.JButton.setEnabled(boolean)> in (FieldSetComponent.java:210)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.setEnabled(boolean)> calls method <javax.swing.JButton.setEnabled(boolean)> in (FieldSetComponent.java:211)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.setEnabled(boolean)> calls method <javax.swing.JButton.setEnabled(boolean)> in (FieldSetComponent.java:213)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.setEnabled(boolean)> calls method <javax.swing.JButton.setEnabled(boolean)> in (FieldSetComponent.java:214)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.setEnabled(boolean)> calls method <javax.swing.JComboBox.setEnabled(boolean)> in (FieldSetComponent.java:207)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.setEnabled(boolean)> calls method <javax.swing.JTextField.setEnabled(boolean)> in (FieldSetComponent.java:204)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.setFields(java.util.Set)> calls constructor <javax.swing.DefaultListModel.<init>()> in (FieldSetComponent.java:280)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.setFields(java.util.Set)> calls method <javax.swing.DefaultListModel.addElement(java.lang.Object)> in (FieldSetComponent.java:282)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.setFields(java.util.Set)> calls method <javax.swing.DefaultListModel.addListDataListener(javax.swing.event.ListDataListener)> in (FieldSetComponent.java:286)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.setFields(java.util.Set)> calls method <javax.swing.JList.setModel(javax.swing.ListModel)> in (FieldSetComponent.java:288)
Method <org.jabref.gui.customentrytypes.FieldSetComponent.setListSelectionMode(int)> calls method <javax.swing.JList.setSelectionMode(int)> in (FieldSetComponent.java:179)
Method <org.jabref.gui.groups.GroupTreeNodeViewModel.expandSubtree(javax.swing.JTree)> calls method <javax.swing.JTree.expandPath(javax.swing.tree.TreePath)> in (GroupTreeNodeViewModel.java:142)
Method <org.jabref.gui.groups.GroupTreeNodeViewModel.getTreePath()> calls constructor <javax.swing.tree.TreePath.<init>([Ljava.lang.Object;)> in (GroupTreeNodeViewModel.java:179)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.mergeEntries()> calls method <javax.swing.JDialog.dispose()> in (MergeSharedEntryDialog.java:111)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showConfirmationDialog()> calls method <javax.swing.JDialog.dispose()> in (MergeSharedEntryDialog.java:102)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showConfirmationDialog()> calls method <javax.swing.JOptionPane.showConfirmDialog(java.awt.Component, java.lang.Object, java.lang.String, int)> in (MergeSharedEntryDialog.java:97)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showMergeDialog()> calls constructor <javax.swing.JButton.<init>(java.lang.String)> in (MergeSharedEntryDialog.java:73)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showMergeDialog()> calls constructor <javax.swing.JButton.<init>(java.lang.String)> in (MergeSharedEntryDialog.java:76)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showMergeDialog()> calls constructor <javax.swing.JLabel.<init>(java.lang.String)> in (MergeSharedEntryDialog.java:67)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showMergeDialog()> calls constructor <javax.swing.JPanel.<init>()> in (MergeSharedEntryDialog.java:79)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showMergeDialog()> calls constructor <javax.swing.border.EmptyBorder.<init>(int, int, int, int)> in (MergeSharedEntryDialog.java:68)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showMergeDialog()> calls method <javax.swing.JButton.addActionListener(java.awt.event.ActionListener)> in (MergeSharedEntryDialog.java:74)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showMergeDialog()> calls method <javax.swing.JButton.addActionListener(java.awt.event.ActionListener)> in (MergeSharedEntryDialog.java:77)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showMergeDialog()> calls method <javax.swing.JDialog.add(java.awt.Component, java.lang.Object)> in (MergeSharedEntryDialog.java:70)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showMergeDialog()> calls method <javax.swing.JDialog.add(java.awt.Component, java.lang.Object)> in (MergeSharedEntryDialog.java:71)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showMergeDialog()> calls method <javax.swing.JDialog.add(java.awt.Component, java.lang.Object)> in (MergeSharedEntryDialog.java:83)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showMergeDialog()> calls method <javax.swing.JDialog.addWindowListener(java.awt.event.WindowListener)> in (MergeSharedEntryDialog.java:85)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showMergeDialog()> calls method <javax.swing.JDialog.pack()> in (MergeSharedEntryDialog.java:92)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showMergeDialog()> calls method <javax.swing.JDialog.setDefaultCloseOperation(int)> in (MergeSharedEntryDialog.java:84)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showMergeDialog()> calls method <javax.swing.JDialog.setMinimumSize(java.awt.Dimension)> in (MergeSharedEntryDialog.java:50)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showMergeDialog()> calls method <javax.swing.JDialog.setVisible(boolean)> in (MergeSharedEntryDialog.java:93)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showMergeDialog()> calls method <javax.swing.JLabel.setBorder(javax.swing.border.Border)> in (MergeSharedEntryDialog.java:68)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showMergeDialog()> calls method <javax.swing.JPanel.add(java.awt.Component, java.lang.Object)> in (MergeSharedEntryDialog.java:80)
Method <org.jabref.gui.shared.MergeSharedEntryDialog.showMergeDialog()> calls method <javax.swing.JPanel.add(java.awt.Component, java.lang.Object)> in (MergeSharedEntryDialog.java:81)

</details

**Comment by Siedlerchr** [comment]
I thought we removed the custom entry types? #4716

**Comment by LinusDietz** [comment]
> I thought we removed the custom entry types? #4716

Yep, just deleted it again

**Comment by Siedlerchr** [comment]
Refs #4178

**Comment by Siedlerchr** [comment]
@tobiasdiez  Could you please tell us if this is obsolete or can be replaced?

```
   Method <org.jabref.gui.groups.GroupTreeNodeViewModel.expandSubtree(javax.swing.JTree)> calls method <javax.swing.JTree.expandPath(javax.swing.tree.TreePath)> in (GroupTreeNodeViewModel.java:142)
    Method <org.jabref.gui.groups.GroupTreeNodeViewModel.getTreePath()> calls constructor <javax.swing.tree.TreePath.<init>([Ljava.lang.Object;)> in (GroupTreeNodeViewModel.java:179)
```

**Comment by tobiasdiez** [comment]
I guess they can be removed as they are no longer needed, but would have to check myself to be sure.

**Comment by Siedlerchr** [comment]
I converted the Icon and the Font stuff, this is the base for https://github.com/JabRef/jabref/issues/4582#issuecomment-487368710 which would enable us to implement the FontSelectorDialog from controlsfx 
FontStyle is now splitted up in Weight and Posture

**Comment by Siedlerchr** [comment]
Accroding to this bug report, we may need to define an exception for the java.awt.desktop class:
https://bugs.openjdk.java.net/browse/JDK-8091107

File: src/main/java/org/jabref/gui/GUIGlobals.java [hunk 1]
```diff
@@ -48,13 +48,12 @@ public static void init() {
...
+        GUIGlobals.currentFont = new Font(Globals.prefs.get(JabRefPreferences.FONT_FAMILY), Globals.prefs.getInt(JabRefPreferences.FONT_SIZE));
     }
 
-    public static void setFont(int size) {
-        currentFont = new Font(currentFont.getFamily(), currentFont.getStyle(), size);
+    public static void setFont(double size) {
+        currentFont = new Font(currentFont.getFamily(), size);
         // update preferences
-        Globals.prefs.putInt(JabRefPreferences.FONT_SIZE, size);
+        Globals.prefs.putDouble(JabRefPreferences.FONT_SIZE, size);
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/GUIGlobals.java]: I guess with this we will run into backwards compatibility issues similar to the ones we had with a different preference option.

**Comment by tobiasdiez** [comment]
At least the open files / browser (and maybe mail program) should work with `getHostServices().showDocument`

File: src/main/java/org/jabref/gui/GUIGlobals.java (hunk 1 above)
Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/GUIGlobals.java]: Hmm, yeah will look if it makes sense to either convert to int or store in new settings

**Comment by Siedlerchr** [comment]
GetHostService only works in application class from what I saw. Would be a bit more complicated to pass it to others

**Comment by Siedlerchr** [comment]
I fixed the remaining swing stuff (except the undo manager).
I checked the usages of awt, but they are something we can't do anything about. 
e.g. one is coming from the pdfbox handler and the other ones are desktop-methods.

File: src/main/java/org/jabref/gui/GUIGlobals.java (hunk 1 above)
Comment (koppor) [reply src/main/java/org/jabref/gui/GUIGlobals.java]: Naive question: can't be stick to Int and just convert to Double during setting the stuff? I would assume that 14.5pt is currently not used --> future work ^^

File: src/main/java/org/jabref/preferences/JabRefPreferences.java [hunk 2]
```diff
@@ -600,7 +600,7 @@ private JabRefPreferences() {
         defaults.put(SHOW_ADVANCED_HINTS, Boolean.TRUE);
         defaults.put(RENAME_ON_MOVE_FILE_TO_FILE_DIR, Boolean.TRUE);
 
-        defaults.put(FONT_STYLE, Font.PLAIN);
+        defaults.put(FONT_STYLE, 0); //for backwards compatibility, is equal to java.awt.Font.PLAIN
```
Comment (LinusDietz) [inline src/main/java/org/jabref/preferences/JabRefPreferences.java]: Is this a merge error?

Comment (Siedlerchr) [reply src/main/java/org/jabref/preferences/JabRefPreferences.java]: No this is correct, see the comment.

**Comment by LinusDietz** [comment]
I would propose to merge this one and create a similar followup PR, but without the exceptions defined here: https://github.com/JabRef/jabref/pull/4894/files#diff-f64d223f20f6b0e53b00c8dbd2be2ce1R23

File: src/main/java/org/jabref/gui/GUIGlobals.java (hunk 1 above)
Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/GUIGlobals.java]: I changed it already to int again.         Globals.prefs.putInt(JabRefPreferences.FONT_SIZE, size);
So it's fine

**Comment by Siedlerchr** [comment]
All swing related code except for the undo manager.

**Pull request merged**